actions/gha-workflow-validator/src/index.ts

```diff
@@ -0,0 +1,72 @@
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import { commentOnPrOrUpdateExisting, deleteCommentOnPRIfExists, getComparison } from "./github.js";
+import { validateActionReferenceChanges } from "./action-reference-validations.js";
+import { filterForGithubWorkflowChanges, formatGithubComment, parseAllAdditions } from "./utils.js";
+import { COMMENT_HEADER } from "./strings.js";
+
+(async () => {
+  const { token, owner, repo, base, head, prNumber } = getInvokeContext();
+  const octokit = github.getOctokit(token);
+
+  const allFiles = await getComparison(octokit, owner, repo, base, head);
```

Comment:
How do you feel about this?

```suggestion
  const changedFiles = await getChangedFiles(octokit, owner, repo, base, head);
```

I'm wondering if we should accept a list of changed files instead, and chain it with something like https://github.com/tj-actions/changed-files

Reply:
`tj-actions/changed-files` will not return the same thing as the Github API `compare` endpoint.

I'm trying to only validate the changes introduced in a PR, and by using the `compare` endpoint you get the diff. I don't think it's reasonable for people have to overhaul a whole file due to linting errors unrelated to their change.

Reply:
Good point, I didnt realize you're doing a line diff rather than a file diff